nemo/collections/nlp/models/dialogue/dialogue_zero_shot_slot_filling_model.py

```diff
@@ -0,0 +1,783 @@
+            'overall_slot_recall': overall_slot_recall,
+            'overall_slot_f1': overall_slot_f1,
+            'unified_slot_precision': unified_slot_precision,
+            'unified_slot_recall': unified_slot_recall,
+            'unified_slot_f1': unified_slot_f1,
+            'unified_slot_joint_goal_accuracy': unified_slot_joint_goal_accuracy,
+        }
+
+    def test_step(self, batch, batch_idx):
+        """
+        Lightning calls this inside the test loop with the data from the test dataloader
+        passed in as `batch`.
+        """
+        return self.validation_step(batch, batch_idx)
+
+    def test_epoch_end(self, outputs):
+        """
+        Called at the end of test to aggregate outputs.
+        :param outputs: list of individual outputs of each test step.
+        """
+        return self.validation_epoch_end(outputs)
+
+    def setup_training_data(self, train_data_config: Optional[DictConfig]):
+        self._train_dl = self._setup_dataloader_from_config(cfg=train_data_config, dataset_split='train')
+
+    def setup_validation_data(self, val_data_config: Optional[DictConfig]):
+        self._validation_dl = self._setup_dataloader_from_config(cfg=val_data_config, dataset_split='dev')
+
+    def setup_test_data(self, test_data_config: Optional[DictConfig]):
+        self._test_dl = self._setup_dataloader_from_config(cfg=test_data_config, dataset_split='test')
+
+    def _setup_dataloader_from_config(self, cfg: DictConfig, dataset_split: str):
+        data_processor = DialogueAssistantDataProcessor(self.data_dir, self.tokenizer, cfg=self.cfg.dataset)
+
+        dataset = DialogueZeroShotSlotFillingDataset(
+            dataset_split,
+            data_processor,
+            self.tokenizer,
+            self.cfg.dataset,  # this is the model.dataset cfg, which is diff from train_ds cfg etc
+        )
+
+        return DataLoader(
+            dataset=dataset,
+            batch_size=cfg.batch_size,
+            shuffle=cfg.shuffle,
+            num_workers=cfg.num_workers,
+            pin_memory=cfg.pin_memory,
+            drop_last=cfg.drop_last,
+            collate_fn=dataset.collate_fn,
+        )
+
+    def update_data_dirs(self, data_dir: str, dialogues_example_dir: str):
+        """
+        Update data directories
+
+        Args:
+            data_dir: path to data directory
+            dialogues_example_dir: path to preprocessed dialogues example directory, if not exists will be created.
+        """
+        if not os.path.exists(data_dir):
+            raise ValueError(f"{data_dir} is not found")
+        self.cfg.dataset.data_dir = data_dir
+        self.cfg.dataset.dialogues_example_dir = dialogues_example_dir
+        logging.info(f'Setting model.dataset.data_dir to {data_dir}.')
+        logging.info(f'Setting model.dataset.dialogues_example_dir to {dialogues_example_dir}.')
+
+    def mask_unused_subword_slots(self, slot_preds, subtokens_mask):
```

Comment:
Description for this function please, also I think you are using this function for both slot_preds and slot_labels so maybe the better name for this first arg is `slots` which can apply to both slot_preds and slot_labels